test/helpers/fuzzing.js

```diff
@@ -0,0 +1,242 @@
+  }
+
+  // Generate the values for those keys (including possible conflicts).
+  for (i = 0; i < keysPerReplication.length; i++) {
+    var keyBatch = keysPerReplication[i]
+    var writeBatch = []
+    for (var j = 0; j < keyBatch.length; j++) {
+      var singleWrite = { key: keyBatch[j], values: [] }
+      var shouldConflict = opts.conflicts && test(opts.conflicts / opts.keys, random)
+      var keyWriters = null
+      if (shouldConflict) {
+        var numConflicts = Math.floor(random() * opts.writers) + 1
+        keyWriters = sample(writers, numConflicts, random, false)
+      } else {
+        keyWriters = sample(writers, 1, random, false)
+      }
+      for (var z = 0; z < keyWriters.length; z++) {
+        var valueString = sample(ALPHABET, opts.valueSize, random, true).join('')
+        singleWrite.values[keyWriters[z]] = valueString
+      }
+      writeBatch.push(singleWrite)
+    }
+    writesPerReplication.push(writeBatch)
+  }
+
+  return writesPerReplication
+}
+
+function validate (t, db, processedBatches, cb) {
+  var expectedWrites = {}
+  // Assuming the batches are insertion order.
+  for (var i = 0; i < processedBatches.length; i++) {
+    var writeBatch = processedBatches[i]
+    for (var j = 0; j < writeBatch.length; j++) {
+      var singleWrite = writeBatch[j]
+      expectedWrites[singleWrite.key] = singleWrite.values.filter(val => !!val)
+    }
+  }
+  var allKeys = Object.keys(expectedWrites)
+
+  t.test(['validating after', processedBatches.length, 'replications'].join(' '), function (t) {
+    t.plan(allKeys.length + 1)
+
+    var readStream = db.createReadStream('/')
+    readStream.on('end', function () {
+      var keys = Object.keys(expectedWrites)
+      t.same(keys.length, 0, 'missing keys: ' + keys)
+      if (keys.length === 0) return cb()
+      return cb(new Error(`missing keys: ${keys}`))
+    })
+    readStream.on('error', cb)
+    readStream.on('data', function (nodes) {
+      if (!nodes) return
+      var key = nodes[0].key
+      var values = nodes.map(node => node.value)
+      t.same(values, expectedWrites[key])
+      delete expectedWrites[key]
+    })
+  })
+}
+
+function generateFailingTest (name, dbCount, writesPerReplication, writeOps, cb) {
+  writeOps.push(err => {
+    t.error(err)
+    t.end()
+  }).toString()
+  fs.writeFile(p.join(__dirname, '..', name + '.js'), prettier.format(`
+    var tape = require('tape')  
+
+    var run = require('./helpers/run')
+    var put = require('./helpers/put')
+    var create = require('./helpers/create')
+
+    var validate = require('./helpers/fuzzing').validate
+
+    var writesPerReplication = ${JSON.stringify(writesPerReplication)}
+
+    tape('autogenerated failing fuzz test', function (t) {
+      create.many(${dbCount}, function (err, dbs, replicateByIndex) {
+        t.error(err)
+        run(${writeOps.map(op => op.toString())})
+      })
+    })`, { singleQuote: true, semi: false }, { encoding: 'utf-8' }, cb))
```

Comment:
i'm unsure if it would be better to simply print the test instead? then this would work with ci